app/assets/javascripts/discourse/tests/integration/components/plugin-outlet-test.gjs

```diff
@@ -1056,3 +1057,80 @@ module(
     });
   }
 );
+
+module(
+  "Integration | Component | plugin-outlet | argument currying",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    test("makes arguments available at top level", async function (assert) {
+      extraConnectorComponent(
+        "test-name",
+        <template>
+          <span class="gjs-test">{{@arg1}} from glimmer</span>
+        </template>
+      );
+
+      await render(
+        <template>
+          <PluginOutlet
+            @name="test-name"
+            @outletArgs={{hash arg1="Hello world"}}
+          />
+        </template>
+      );
+      assert.dom(".glimmer-test").hasText("Hello world from glimmer");
+    });
+
+    test("makes arguments available at top level in classic components", async function (assert) {
+      extraConnectorComponent(
+        "test-name",
+        class extends ClassicComponent {
+          <template>
+            <span class="classic-test">{{this.arg1}} from classic</span>
+          </template>
+        }
+      );
+
+      await render(
+        <template>
+          <PluginOutlet
+            @name="test-name"
+            @outletArgs={{hash arg1="Hello world"}}
+          />
+        </template>
+      );
+      assert.dom(".classic-test").hasText("Hello world from classic");
+    });
+
+    test("makes arguments available at top level in classic components", async function (assert) {
```

Comment:
this one should describe the name-clashing logic?

Reply:
fixed